inventory-generation/identity-management/main.yml

```diff
@@ -61,17 +65,43 @@
     set_fact:
       claim_content: "{{ { 'customer_name': customer_name, 'project_name': project_name, 'ipa_validate_certs': ipa_validate_certs, 'ipa_host': ipa_host, 'ipa_admin_user': ipa_admin_user, 'ipa_admin_password': ipa_admin_password, 'lodestar_identities': { 'users': users, 'groups': usrgrp } } | to_nice_yaml(indent=2) }}"
 
+  - name: "Check For Existing Inventory File"
+    stat:
+      path: "{{ directory }}/iac/inventories/identity-management/inventory/group_vars/all.yml"
+    register: existing_inv
+
+  - name: "Pull existing inventory vars"
+    include_vars:
+      file: "{{ directory }}/iac/inventories/identity-management/inventory/group_vars/all.yml"
+      name: existing_inv_contents
+    when:
+      - existing_inv.stat.exists
+
+  - name: "Diff existing vars with new inventory"
+    set_fact:
+      inv_has_diff: true
+    when:
+      - existing_inv.stat.exists
+      - existing_inv_contents is defined
+      - existing_inv_contents != (claim_content | from_yaml)
+
   - name: "Write inventory to file"
     copy:
       content: "{{ claim_content }}"
       dest: "{{ directory }}/iac/inventories/identity-management/inventory/group_vars/all.yml"
+    when:
+      - (inv_has_diff is defined and inv_has_diff is true) or not existing_inv.stat.exists
```

Comment:
I believe this conditional can be removed, the `copy` module should not create/update a file if the content hasn't changed or did you observe different behavior when testing?

Reply:
🤷‍♂️  I think this was more of a safety blanket for me haha. I can go ahead and remove this though.

Reply:
@pabrahamsson Ah wait. I remember why I did this. These checks are more logic needed to know when to create a new file (because we're creating unique timestamped files now instead of just the idmgmt.yml, etc. So I think these should be good to leave in, but let me know if you disagree.

Reply:
But isn't that only true for the resourceclaim, we aren't creating timestamped files in the inventory?